internal/models/server.go

```diff
@@ -29,11 +29,20 @@ type Server struct {
 	MultiHop    bool         `json:"multihop,omitempty"`
 	WgPubKey    string       `json:"wgpubkey,omitempty"`
 	Free        bool         `json:"free,omitempty"`
-	Stream      bool         `json:"stream,omitempty"`
+	Stream      bool         `json:"stream,omitempty"` // TODO move to Features in v4
 	Premium     bool         `json:"premium,omitempty"`
 	PortForward bool         `json:"port_forward,omitempty"`
 	Keep        bool         `json:"keep,omitempty"`
 	IPs         []netip.Addr `json:"ips,omitempty"`
+	Features    *Features    `json:"features,omitempty"`
+}
+
+type Features struct {
+	SecureCore bool `json:"secure_core,omitempty"`
+	Tor        bool `json:"tor,omitempty"`
+	P2P        bool `json:"p2p,omitempty"`
+	Streaming  bool `json:"stream,omitempty"`
+	IPv6       bool `json:"ipv6,omitempty"`
```

Comment:
What's the advantage over having them directly as fields of `Server` as they were?
I see it makes it a bit tidier indeed, but to keep it simple/retro-compatible/no-change-needed, I would keep the same structure as before. You can however add a `// TODO v4 ...` comment describing it would be nice to have them in a Features struct though.

Reply:
it was mostly not to add too many parameters to the [iptoserver.go add](https://github.com/qdm12/gluetun/pull/2182/files#diff-24ccd1d540dfbe5e9a4e94c5fb8e12ac3ffa3bc0a425a2cd0d70557c51b2eefdR12-R13) function. I'm also fine with just making it part of the server struct though.

Reply:
Let's have these new fields directly in the Server struct, and add an unexported `features` (no first capital letter for unexported types) struct in the updater package to keep the code as clean as it is 😉